### PR TITLE
test(e2e): give the runner extra time to wait for a killed node

### PR DESCRIPTION
**Pull request by andynog**
One reason some e2e tests are failing is that the kill perturbation has a short timeout value of 20 seconds, causing waitForNode to time out. Doubling this timeout improves test stability and helps ensure the tests pass

[test failure](https://github.com/cometbft/cometbft/actions/runs/11546210646/job/32134226559)

#### PR checklist

- [ ] Tests written/updated
- [x] Changelog entry added in `.changelog` (we use [unclog](https://github.com/informalsystems/unclog) to manage our changelog)
- [ ] Updated relevant documentation (`docs/` or `spec/`) and code comments

### CODE REVIEW

**Review by melekes — APPROVED** [review]
Thanks @andynog ❤️

File: test/e2e/runner/perturb.go [hunk 1]
```diff
@@ -64,6 +64,10 @@ func PerturbNode(ctx context.Context, node *e2e.Node, perturbation e2e.Perturbat
 		}
 		if node.PersistInterval == 0 {
 			timeout *= 5
+		} else {
```
Comment (cason) [inline test/e2e/runner/perturb.go:67]: What is the rationale of this if?

I assume that when `PersistInterval == 0`, the application has no state so it should restart from genesis, is that right?

File: test/e2e/runner/rpc.go [hunk 2]
```diff
@@ -92,9 +92,11 @@ func waitForNode(ctx context.Context, node *e2e.Node, height int64, timeout time
 			return nil, ctx.Err()
 		case <-timer.C:
 			status, err := client.Status(ctx)
+			sinceLastChanged := time.Since(lastChanged)
```
Comment (cason) [inline test/e2e/runner/rpc.go:95]: You could just call `time.Since` again in the log message, but ok.

**Review by cason — APPROVED** [review]
We could maybe increase the standard timeout to 30s.

File: test/e2e/runner/perturb.go (hunk 1 above)
Comment (andynog) [reply test/e2e/runner/perturb.go:67]: to be honest, just looking at the PersistInterval field, I only see it used in validation and perturb logic but in `if` clauses, so I think this field might not have logic that drives the state persistence. We can revisit it in the future, my PR just adds more time to the timeout in this case.

**Pull request merged**